accountgw_test.go

```diff
@@ -26,58 +27,75 @@ func TestGatewayAccount_EstimateAndExecute(t *testing.T) {
 	testSet := map[string][]testSetType{
 		"devnet": {{
 			ExecuteCalls: []types.FunctionCall{{
-				EntryPointSelector: "increment",
-				ContractAddress:    types.StrToFelt(testConfig.CounterAddress),
+				EntryPointSelector: types.GetSelectorFromNameFelt("increment"),
 			}},
 			QueryCall: types.FunctionCall{
-				EntryPointSelector: "get_count",
-				ContractAddress:    types.StrToFelt(testConfig.CounterAddress),
+				EntryPointSelector: types.GetSelectorFromNameFelt("get_count"),
 			},
 		}},
 		"testnet": {{
 			ExecuteCalls: []types.FunctionCall{{
-				EntryPointSelector: "increment",
-				ContractAddress:    types.StrToFelt(testConfig.CounterAddress),
+				EntryPointSelector: types.GetSelectorFromNameFelt("increment"),
 			}},
 			QueryCall: types.FunctionCall{
-				EntryPointSelector: "get_count",
-				ContractAddress:    types.StrToFelt(testConfig.CounterAddress),
+				EntryPointSelector: types.GetSelectorFromNameFelt("get_count"),
 			},
 		}},
 	}[testEnv]
 
 	for _, test := range testSet {
-		// shim a keystore into existing tests.
+		// Convert the counter address to a Felt.
+		counterAddress, err := utils.HexToFelt(testConfig.CounterAddress)
+		if err != nil {
+			t.Fatalf("Failed to convert counter address to Felt: %v", err)
+		}
+
+		// Set the contract address for the execute and query calls.
+		for i := range test.ExecuteCalls {
+			test.ExecuteCalls[i].ContractAddress = counterAddress
+		}
```

Comment:
I think we can move this to the start of the function. It'll also save us ranging over the test.ExecuteCalls

Reply:
You mean something like this instead?

```
	counterAddress, err := utils.HexToFelt(testConfig.CounterAddress)
	if err != nil {
		t.Fatalf("Failed to convert counter address to Felt: %v", err)
	}


	testSet := map[string][]testSetType{
		"devnet": {{
			ExecuteCalls: []types.FunctionCall{{
				EntryPointSelector: types.GetSelectorFromNameFelt("increment"),
				ContractAddress: counterAddress,
			}},
			QueryCall: types.FunctionCall{
				EntryPointSelector: types.GetSelectorFromNameFelt("get_count"),
				ContractAddress: counterAddress,
			},
		}},
		
```

Reply:
Yes exactly